### PR TITLE
Prevent scripts from loading if behaviors are not used

### PR DESCRIPTION
**Pull request by michalczaplinski**
## What?
When no behaviors are used, do not load the `interactivity-view.js` file.

## Why?
So that we don't wastefully load JS.

## How?
Adding a workaround very similar to the one [currently used](https://github.com/WordPress/gutenberg/blob/5ec913ff88158a13837cb88fdf3c7effb2f7aa05/packages/block-library/src/navigation/index.php#L648-L660) in the Navigation block.

## Testing Instructions
1. Open a Post or Page.
2. Insert an Image Block.
3. Open the post preview and make sure that the `interactivity-view.js` file is not loaded.
4. In the **Advanced** panel select the `Lightbox` behavior and save the post.
5. Go back to the preview and make sure that this time the file IS being loaded.
6. Try the same thing with more than one Image block.

### CODE REVIEW

File: packages/block-library/src/image/index.php [hunk 1]
```diff
@@ -30,14 +31,31 @@ function render_block_core_image( $attributes, $content ) {
 		$processor->set_attribute( 'data-id', $attributes['data-id'] );
 	}
 
+	$should_load_view_script = ( array_key_exists( 'behaviors', $attributes ) &&
```
Comment (artemiomorales) [inline packages/block-library/src/image/index.php]: I believe we need to add a check not just at the block attribute level, but at the theme level.

**Review by artemiomorales — CHANGES_REQUESTED** [review]
Hi @michalczaplinski! I took a look and this PR works if the theme.json default value for the behavior is false, but it breaks the lightbox if the default value is true. To Illustrate further:

**This works**
```
"behaviors": {
	"blocks": {
		"core/image": {
			"lightbox": {
				"enabled": false
			}
		}
	}
}
```

**This doesn't work**

```
"behaviors": {
	"blocks": {
		"core/image": {
			"lightbox": {
				"enabled": true
			}
		}
	}
}
```

File: packages/block-library/src/image/index.php (hunk 1 above)
Comment (SantosGuillamot) [reply packages/block-library/src/image/index.php]: Yes, we should do that. I guess we should add all the checks we are using for the lightbox. For example, we are also checking if the image is a link, right? If I am not mistaken, if it is a link we shouldn't load the scripts either.

Comment (michalczaplinski) [reply packages/block-library/src/image/index.php]: You're absolutely right! Sorry about this omission. I've updated the PR now with the logic that is replicated from [`block-supports/behaviors.php`](https://github.com/WordPress/gutenberg/blob/2c92b02fcf5d0911fc1355d910b276fde48195e2/lib/block-supports/behaviors.php#L48-L60).

Comment (artemiomorales) [reply packages/block-library/src/image/index.php]: Ok great, tested and this seems to be working now 👍

**Review by artemiomorales — COMMENTED** [review]
@michalczaplinski As I think on it, I believe we should also add a test for this.

**Comment by michalczaplinski** [comment]
Some quick comments on code & tests:


https://github.com/WordPress/gutenberg/assets/5417266/fe30bf6b-ccf3-46a9-baff-8fb59db3e0ad

**Review by artemiomorales — COMMENTED** [review]
@michalczaplinski  It's working — looking good! I think we should also add the following test cases:
-Should add view script if lightbox is enabled via theme.json
-Should NOT add view script if lightbox is enabled via theme.json but overriden in block settings

**Review by artemiomorales — APPROVED** [review]
@michalczaplinski I've added the additional test cases. Let me know if all looks good to you! If so, I think we can merge.

**Review by cbravobernal — APPROVED** [review]
Looks good to me too.

**Comment by michalczaplinski** [comment]
Looks good, thanks for adding the extra tests @artemiomorales !

**Pull request merged**